tests/components/rest/test_sensor.py

```diff
@@ -677,3 +679,50 @@ def test_update_request_exception(self, mock_req):
         """Test update when a request exception occurs."""
         self.rest.update()
         assert self.rest.data is None
+
+
+async def test_reload(hass):
+    """Verify we can reload reset sensors."""
+
+    with requests_mock.Mocker() as mock_req:
+        mock_req.get("http://localhost", text="test data")
+        await async_setup_component(
+            hass,
+            "sensor",
+            {
+                "sensor": {
+                    "platform": "rest",
+                    "method": "GET",
+                    "name": "mockrest",
+                    "resource": "http://localhost",
+                }
+            },
+        )
+        await hass.async_block_till_done()
+        await hass.async_start()
+        await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 1
+
+    assert hass.states.get("sensor.mockrest")
+
+    yaml_path = path.join(
+        _get_fixtures_base_path(), "fixtures", "rest/configuration.yaml",
+    )
+    with patch.object(
+        hass_config, "YAML_CONFIG_FILE", yaml_path
+    ), requests_mock.Mocker() as mock_req:
+        mock_req.get("http://localhost", text="test data 2")
+        await hass.services.async_call(
+            "rest", SERVICE_RELOAD, {}, blocking=True,
+        )
+        await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 1
+
+    assert hass.states.get("sensor.mockreset") is None
+    assert hass.states.get("sensor.rollout")
+
+
+def _get_fixtures_base_path():
+    return path.dirname(path.dirname(path.dirname(__file__)))
```

Comment:
We have this test helper:
https://github.com/home-assistant/core/blob/eaac00acfcaa1dac1286eb184416a6bed17a3238/tests/common.py#L328-L332

Reply:
Nevermind, I see that we don't want to read the file here.